backend/src/test/java/hu/flowacademy/vajdasagbrand/controller/UserControllerTest.java

```diff
@@ -0,0 +1,68 @@
+package hu.flowacademy.vajdasagbrand.controller;
+
+import hu.flowacademy.vajdasagbrand.dto.UserDTO;
+import hu.flowacademy.vajdasagbrand.persistence.entity.Type;
+import hu.flowacademy.vajdasagbrand.repository.UserRepository;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import lombok.extern.slf4j.Slf4j;
+import com.github.javafaker.Faker;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static hu.flowacademy.vajdasagbrand.helpers.UserHelper.*;
+import static io.restassured.RestAssured.given;
+
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Slf4j
+class UserControllerTest {
+
+
+    @LocalServerPort
+    private int port;
+    private static final Faker faker = new Faker();
+    private static UserDTO userDTO;
+    private static UserRepository userRepository;
+
+
+    @BeforeEach
+    private void setUp() {
+        RestAssured.port = port;
+    }
+
+    @BeforeAll
+    private static void beforeAll() {
+        userDTO = UserDTO.builder()
+                .id(faker.gameOfThrones().dragon())
```

Comment:
Ne a sárkányok neve legyen az id! Sőt, ne is legyen id beállítva, illetve szervezd át a UserDTO builder hívást a reigsztrációs tesztbe